src/main/java/com/modernjava/demo/controller/ModernJavaController.java

```diff
@@ -37,7 +41,10 @@ public String filtered(@PathVariable String filterType) {
 	 * Filtering apples the Java 8 way!
 	 */
 	@GetMapping(path = "/filteredJ8/{filterType}")
-	public String filteredJ8(@PathVariable String filterType) {
-		return appleService.filterApplesJ8(filterType);
+	public ResponseEntity<List<Apple>> filteredJ8(@PathVariable
```

Comment:
What about the other methods in the controller that are still returning String. It might be idea to update to return ResponseEntity too.
Also, if you give an invalid path parameter you get back the unfiltered and unsorted list. Is this correct behavior would be better to return an error message. And provided an additional end point if  client wish to see unadulterated list.

Reply:
Thanks for the suggestion, yes I agree.
I'll update the pull request.